workspaces/ui-v2/src/pages/docs/components/ShapeEditor.tsx

```diff
@@ -0,0 +1,465 @@
+import React, { FC, useMemo } from 'react';
+import {
+  makeStyles,
+  darken,
+  lighten,
+  IconButton,
+  ButtonGroup,
+  Button,
+  TextField,
+} from '@material-ui/core';
+import { Check as CheckIcon } from '@material-ui/icons';
+import ClassNames from 'classnames';
+import Color from 'color';
+import DeleteOutlineIcon from '@material-ui/icons/DeleteOutline';
+
+import { IFieldDetails, IShapeRenderer } from '<src>/types';
+import { JsonType } from '@useoptic/optic-domain';
+import * as Theme from '<src>/styles/theme';
+
+export const ShapeEditor: FC<{
+  fields: IFieldDetails[];
+  selectedFieldId: string | null;
+  setSelectedField: (fieldId: string | null) => void;
+}> = ({ fields, selectedFieldId, setSelectedField }) => {
+  const classes = useStyles();
+
+  let onChangeFieldDescription = useMemo(
+    () => (fieldId: string, description: string) => {
+      console.log('description changed for field', fieldId, description);
+    },
+    []
+  );
+
+  return (
+    <div className={classes.container}>
+      {fields.length > 0 && (
+        <ul className={classes.rowsList}>
+          {fields.map((field) => (
+            <li className={classes.rowListItem}>
+              <Row
+                field={field}
+                description={''} // TODO FLEB: wire me in
+                selected={selectedFieldId === field.fieldId}
+                onChangeDescription={onChangeFieldDescription}
+                onSelect={setSelectedField}
+              />
+            </li>
+          ))}
+        </ul>
+      )}
+    </div>
+  );
+};
+
+const Row: FC<{
+  field: IFieldDetails;
+  description: string;
+  selected: boolean;
+
+  onChangeDescription?: (fieldId: string, description: string) => void;
+  onSelect?: (fieldId: string) => void;
+}> = function ShapeEditorRow({
```

Comment:
Any reason you set the Row variable to a named function? I think just naming `const ShapeEditorRow` sets the name of the function, and makes it easier to trace function calls between things

Reply:
Short names for internal use, while more explicit names for the debugger. Ever since fat arrow became a thing people don't use named functions enough anymore, they are so much nicer to debug rather than a massive stack trace of "anonymous". Or maybe debuggers have caught up recently and resolve variable names? I'm not sure, just a bit of a habit. 🤷‍♂️